From 827abc36e1c21775c23f6b180d86c08586ca1661 Mon Sep 17 00:00:00 2001
From: adam radford <aradford@gmail.com>
Date: Wed, 11 May 2011 18:34:17 -0700
Subject: [PATCH 48/53] [SCSI] megaraid_sas: Disable interrupts/free_irq() in  megasas_shutdown()

commit 46fd256e05581e4a04d5a8ec107d35afe938c65b upstream.

The following patch for megaraid_sas disables interrupts and
free_irq() in megasas_shutdown().

Signed-off-by: Adam Radford <aradford@gmail.com>
Signed-off-by: James Bottomley <jbottomley@parallels.com>
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 drivers/scsi/megaraid/megaraid_sas_base.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/scsi/megaraid/megaraid_sas_base.c b/drivers/scsi/megaraid/megaraid_sas_base.c
index e682f1ab09df..245378009ad7 100644
--- a/drivers/scsi/megaraid/megaraid_sas_base.c
+++ b/drivers/scsi/megaraid/megaraid_sas_base.c
@@ -4526,6 +4526,11 @@ static void megasas_shutdown(struct pci_dev *pdev)
 	instance->unload = 1;
 	megasas_flush_cache(instance);
 	megasas_shutdown_controller(instance, MR_DCMD_CTRL_SHUTDOWN);
+	instance->instancet->disable_intr(instance->reg_set);
+	free_irq(instance->msi_flag ? instance->msixentry.vector :
+		 instance->pdev->irq, instance);
+	if (instance->msi_flag)
+		pci_disable_msix(instance->pdev);
 }
 
 /**
-- 
1.8.0


